File: Exercism Codes/JavaScript/resistor-color-trio.js
export class ResistorColorTrio {
  constructor(colors) {
    const COLORS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"];

    if (colors.some(c => !COLORS.includes(c))) {
      throw new Error("invalid color");
    }

    const numStr = String(COLORS.indexOf(colors[0])) + COLORS.indexOf(colors[1]) + "0".repeat(COLORS.indexOf(colors[2]));
    let num = parseInt(numStr);
    let unit;
    if (num >= 1000000000) {
      num /= 1000000000;
      unit = "gigaohms";
    }
    else if (num >= 1000000) {
      num /= 1000000;
      unit = "megaohms";
    }
    else if (num >= 1000) {
      num /= 1000;
      unit = "kiloohms";
    }
    else {
      unit = "ohms";
    }
    this._label = `Resistor value: ${num} ${unit}`;
  }

  get label() {
    return this._label;
  }
}

File: Exercism Codes/C#/ResistorColorTrio.cs
public static class ResistorColorTrio
{
    private static readonly List<string> COLORS = [
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
    ];

    public static string Label(string[] colors)
    {
        string str = COLORS.IndexOf(colors[0]) + "" + COLORS.IndexOf(colors[1]) + new string('0', COLORS.IndexOf(colors[2]));

        long value = long.Parse(str);

        if (value < 1000)
        {
            return $"{value} ohms";
        }
        else if (value < 1_000_000)
        {
            return $"{value / 1000} kiloohms";
        }
        else if (value < 1_000_000_000)
        {
            return $"{value / 1_000_000} megaohms";
        }
        else
        {
            return $"{value / 1_000_000_000} gigaohms";
        }
    }
}
